relay-server/src/http.rs

```diff
@@ -39,51 +38,72 @@ impl HttpError {
             Self::Reqwest(error) => error.is_timeout(),
             Self::Json(_) => false,
             HttpError::Overflow => false,
-            HttpError::NoCredentials => false,
         }
     }
 }
 
 pub struct Request(pub reqwest::Request);
 
 pub struct RequestBuilder {
-    builder: reqwest::RequestBuilder,
+    builder: Option<reqwest::RequestBuilder>,
```

Comment:
This is not actually optional. If there's another safe way to move the request builder, I'd prefer that but I haven't found one.

Reply:
Yeah this is a bit unfortunate, but any reason our `RequestBuilder` can't keep taking `self`?

I sketched this up here: https://gist.github.com/Dav1dde/f10c001462b0df03acfc241c16a6d697

Reply:
This is exactly the kind of signature I wanted to avoid, as it requires to return the builder from the trait method. It's what we had before. Do you prefer that signature?

Reply:
No not necessarily, probably nicer overall with the `&mut` except for the builder itself.